backend/routes.js

```diff
@@ -74,7 +74,24 @@ route.GET('/eventsBefore/{before}/{limit}', {}, async function (request, h) {
     if (!limit) return Boom.badRequest('missing limit')
     if (isNaN(parseInt(limit)) || parseInt(limit) <= 0) return Boom.badRequest('invalid limit')
 
-    const stream = await sbp('backend/db/streamEntriesBefore', before, limit)
+    const stream = await sbp('backend/db/streamEntriesBefore', before, parseInt(limit))
+    request.events.once('disconnect', stream.destroy.bind(stream))
+    return stream
+  } catch (err) {
+    return logger(err)
+  }
+})
+
+route.GET('/eventsBetween/{startHash}/{endHash}', {}, async function (request, h) {
+  try {
+    const { startHash, endHash } = request.params
+    const { offset } = request.query
+
+    if (!startHash) return Boom.badRequest('missing startHash')
+    if (!endHash) return Boom.badRequest('missing endHash')
+    if (offset && (isNaN(parseInt(offset)) || parseInt(offset) < 0)) return Boom.badRequest('invalid offset')
+
+    const stream = await sbp('backend/db/streamEntriesBetween', startHash, endHash, parseInt(offset || 0))
```

Comment:
`parseInt(offset)` is called 3 times here. DRY says it should only be called once.

So, something like this:

```js
const offset = parseInt(request.query.offset || 0)

if (!startHash) return Boom.badRequest('missing startHash')
if (!endHash) return Boom.badRequest('missing endHash')
if (isNaN(offset) || offset < 0) return Boom.badRequest('invalid offset')

const stream = await sbp('backend/db/streamEntriesBetween', startHash, endHash, offset)
```

Reply:
Nice catch. Resolved.
Since the expected type of parseInt function is `string`.
I would like to use this.

`const offset = parseInt(request.query.offset || '0');`